### DISCUSSION TITLE
Abgastemperatur erfassen?

### DISCUSSION THREAD

**Discussion by Kenaschon** [dewenni/ESP_Buderus_KM271#55] [Ideas]
Gibt es eine Möglichkeit, das die Abgastemperatur mit einem Temp-Sensor erfasst werden kann?

Grüße
Manfred

**Comment by dewenni** [comment]
Grundsätzlich sollte das möglich sein.
Auf dem original KM271 ist dafür ein Anschluss vorhanden und auch die Platine von Daniel hat die Analogeingänge mit dabei.
Ich weiß aber nicht welchen Fühler man braucht und wie man ggf. die Schaltung anpassen muss.

**Comment by Kenaschon** [comment]
Gefühlt würde ich einfach den DS18B20 nehmen. Der ist sehr weit verbreitet. Plus und Minus an den Sensor. Den dritten Anschluß an einen digitalen Eingang - fertig.

**Comment by Kenaschon** [comment]
https://espeasy.readthedocs.io/en/latest/Plugin/P004_DS18b20.html

**Comment by Kenaschon** [comment]
Oder diesen hier mit einem Thermoelement. Für Temperaturen über 125°C. Wobei der DS18b20 auch etwas höher als 125°C kann. Er wird dann nur etwas ungenauer.

https://eckstein-shop.de/AdafruitThermocoupleAmplifierwith1-WireBreakoutBoard-MAX31850K

mit diesem

https://www.thermagmbh.de/shop/produkt/motorsport/wassertemperatursensor/einschraub-te-teflon-typ-k/

und fertig.

**Reply by the78mole** [reply]
Normalerweise nimmt man einen NTC 100K mit entsprechend temperaturfester Leitung, den kann die Buderus über den Anschluss auf meiner Platine (mit den ensprechenden Bauteilen bestückt) direkt verwenden. Den kann man sich beim Sensorshop24 zusammenkonfigurieren. 

Ich glaube, der hier passt mechanisch ganz gut: https://www.sensorshop24.de/einschraubtemperaturfuehler-mit-g1-4-gewinde

Optionen: 100K NTC, Leitung PFA (Silikon könnte etwas knapp sein) und entsprechende Leitungslänge wählen.

**Comment by Kenaschon** [comment]
Kannst Du sagen, ob das noch etwas wird?

Grüße

**Reply by dewenni** [reply]
Hallo @Kenaschon ,

wie Daniel schon schrieb, kannst du den Abgasfühler auch völlig unabhängig meiner Software nachrüsten.
Der Sensor wird auch im Original direkt von der Logamatic ausgewertet. Meine Software liest den Wert bereits aus, nur kommt da ohne angeschlossenen Sensor natürlich kein ordentlicher Wert.

<img width="469" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/72ef838a-08e8-442a-890e-6d5a01a0b975">


Auf der original KM271 gibt es für den Sensor den Anschluss FG. Der geht dann direkt auf die Logamatic
<img width="596" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/3371abf9-0183-4275-846b-3413230bb2f6">

<img width="501" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/ab5a8a61-f6c4-4ced-bb50-94c0721536c0">

<img width="590" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/3e4ea6f1-ef3b-4389-a839-62c32231308b">
  

Gewinde müsste ein 1/2" sein. Daher müsste dieser Sensor passen wenn man ihn entsprechend konfiguriert:
https://www.sensorshop24.de/temperaturfuehler-passiv/einschraubfuehler/einschraubtemperaturfuehler-mit-g1-2-gewinde?gad_source=1&gclid=Cj0KCQiA5fetBhC9ARIsAP1UMgHBH_FEjIY2iQBjYeZzoSQhhUNF8mcrJV_3jVyngMlYt-NN3MO7hmwaAr5uEALw_wcB


und Daniel hat das auf seinen Board´s auch vorgesehen. Der Stecker ist aber unter umständen nicht bestückt.
<img width="619" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/4dc1667d-afe6-47b8-9c62-cad563dff28c">

Du müsstest dir also nur einen passenden Sensor kaufen und ggf. das Board von Daniel (sofern du das einsetzt) noch fertig bestücken. Dann sollte die Logamatic selbst den Sensor auswerten und du müsstest automatisch korrekte Werte in der Software sehen.

Wenn du das erfolgreich umgesetzt hast, dann berichte gerne hier darüber.

**Comment by Kenaschon** [comment]
Ich habe mir jetzt mal den Sensor bestellt. Ich werde berichten.

Grüße

**Comment by Iam-klm** [comment]
Hi zusammen.

Bei mir läuft die Abgasanzeige und die MQTT Werte sind schon eine Weile da.
Weil ich zwei Bud's mit der KM271 von Daniel habe, habe ich bei mir bei dem großen Online Händler den jeder kennt A....., folgendes bestellt:
 
für 10,70€ - 2 Stück G 1/2 Zoll Thermowell 1/2" Edelstahl Tauchhülse für Temperatursensor Temperaturfühler (8 x 50 mm)

und

für 8,99€ - 5 Stück WINSINN HT-NTC100K Thermistor 3 x 15 3950 Verdrahtung Buchse 2-Pin XH2.54 Kopf NTC für Sensor Heizblock Hotend 3D-Drucker 
(Temperaturmessung kann bis zu 350 Grad betragen, Kabellänge 1,2m reicht locker bis zum Abgasrohr)
wie man sie beim 3D Drucker einsetzt.

 Somit habe ich sogar noch drei NTC100 als Reserve.

Da meine Boards noch nicht voll bestückt waren habe ich noch:
R17   3K3
R11   100k
C11   100n
C12   33n
R35   0
R39   0
nachbestückt und alles war gut.

![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/142657225/39ad1871-7a68-4322-9374-e0301beeec2a)

**Reply by dewenni** [reply]
@Iam-klm kannst du bei Gelegenheit mal ein Foto machen wo du den Sensor montiert hast?
Mein Abgasrohr ist bestimmt 1,8m lang, bevor es in den Kamin geht. Je weiter man den Sensor Richtung Heizung montiert, desto höher werden die Temperaturen sein die man misst.
Ist bekannt ob die Heizung auf diese Messwerte irgendwie reagiert? Oder dient das nur der Anzeige?
Meine Heizung läuft seit Jahren sehr stabil und ohne Störungen. Ich will sie mit einem zusätzlichen Sensor nicht irgendwie negativ beeinflussen :-)

**Comment by joeflower** [comment]
> @Iam-klm  Ist bekannt ob die Heizung auf diese Messwerte irgendwie reagiert? Oder dient das nur der Anzeige? Meine Heizung läuft seit Jahren sehr stabil und ohne Störungen. Ich will sie mit einem zusätzlichen Sensor nicht irgendwie negativ beeinflussen :-)

Das würde mich auch interessieren. Die Anzeige an sich wäre schon interessant. Aber wenn es die Heizung "irritiert"... :-(

**Comment by Kenaschon** [comment]
Ich habe mal Buderus gefragt.

Anzwort:

"Sehr geehrter Herr Glanemann,
danke für Ihre Anfrage zur Regelung 2017!
Mit dem Modul KM 271 und einem Abgastemperaturfühler ist die Abgastemperaturmessung möglich.
Sie kann im Display abgefragt werden. Überschreitet die Abgastemperatur den eingestellten Grenzwert, wird über das Display eine Fehlermeldung angezeigt. Der Heizkessel sollte dann gewartet werden.
Sind das Modul und der Abgastemperaturfühler installiert, muss die Abgastemperaturmessung aktiviert werden. Der Kessel bleibt auch nach einem Überschreiten des eingestellten Grenzwertes in Betrieb.
Die Werkseinstellung für die Abgastemperaturmessung ist „AUS“.

Der Anschluss hat also keine Auswirkungen auf die Heizungssteuerung, sondern dient zunächst lediglich der Information.
(siehe auch [Serviceanleitung zur R2107](https://buderus-de-de.boschtt-documents.com/download/pdf/file/63020501.pdf), auf Seite 14)

Mit freundlichen Grüßen
Stephan Funke

Buderus Kundendienst, Technische Beratung
Bosch Homecomfort GmbH
Sophienstraße 30-32
35576 Wetzlar
Telefon: 0800 020 3000
E-Mail:    [Kundendienst@Buderus.de](mailto:Kundendienst@Buderus.de) 
Internet:  [www.buderus.de](https://github.com/dewenni/ESP_Buderus_KM271/discussions/www.buderus.de) 

Bosch Homecomfort GmbH
Thermotechnology"

Also alles in Ordnung.

**Reply by dewenni** [reply]
Danke für die Recherche. Dann werde ich mir wohl auch mal einen Sensor bestellen und anschließen.

**Comment by FVerhamme** [comment]
Hallo zusammen, ich hänge mich mal hier dran ;-)
Ich stelle mir gerade die Frage, wohin genau Ihr den Sensor platziert habt.
Generell soll er sich ja am heißesten Punkt des Abgasstroms im Querschnitt des Abgasrohres, des Verbindungsstückes zwischen Brennkammer und Schornstein befinden.
Dort habe ich ein Metallrohr (genauer gesagt ein 90° Bogen) als Abgasrohr.
Mein Sensor hat ein 1/2" Anschlussgewinde. Das Rohr selbst hat keinen definierten Punkt, an dem ich den Sensor einschrauben könnte und in das 1mm dickwandige Rohr ein 1/2" Gewinde schneiden ist ziemlich Sinnbefreit.
Daher also die Frage wie habt Ihr das gelöst.
Ich könnte einfach ein entsprechend großes Loch in das Rohr bohren, den Sensor durchstecken und mit einer 1/2" Mutter festschrauben.
Mein Kessel ist ein Buderus Logano G115 falls das wichtig sein sollte.

P.S. Der Schornsteinfeger misst die Temperatur etwa 1m hinter dem Übergang vom Brenner zum Abgasrohr, weil er dort am einfachsten rankommt ohne sich zu bücken ;-)
Das ist aber eigentlich nicht wirklich korrekt.

Gruß Frank

**Reply by the78mole** [reply]
Hallo Frank,

also ich habe das bestehende Loch, das auch der Schornsteinfeger nutzt (ca. 30 cm vom Kesselstutzen entfernt), mit einem Stufenbohrer so weit aufgebohrt, dass das Sensor-Gewinde sich ausreichend einklemmt (siehe Foto). Man könnte nun noch einen kleinen Einschnitt machen und etwas gegenläufig aufbiegen, so dass man auch das Gewinde eindrehen kann, aber mir recht es, wie es jetzt ist. Der Schornsteinfeger hat damit kein Problem, da dessen Sonde eine "Gummi-Stufenkegel-Abdichtung" hat, die sich an jede Bohrung (bis geschätzt 50 mm) anpasst.

![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/7723396/b14cb137-401a-48b7-ae51-a237f4095e32)

Auch sollte man erwähnen, dass die Kennlinie der NTCs (anders als z.B. bei PT100/500/1000) nicht standardisiert ist. Während es für 25°C genau passt, sollte man nicht erwarten, dass es bei höheren Temperaturen noch stimmt, was die Heizung anzeigt. Aber das ist mir auch ziemlich egal, solange es in der passenden Größenordnung ist. Ich schätze, bei 120°C Anzeige liege ich tatsächlich bei 140-150°C Abgas.

Noch anzumerken ist, dass das Abgas nicht wirklich viel Temperatur verlieren kann. Zwar ist die Strömung im Abgasrohr mit großer Sicherheit turbulent...

Rohrdurchmesser: 100 mm => 0,1 m => Querschnitt 0,007 m²
Abgasdichte: 0,75 kg/m³ (@150°C)
Kesselleistung: 21 kW => 2 kg/h Öl + 33 kg/Luft = 35 kg Abgas pro Stunde
Abgasvolumen: 35 kg/h / 0,75 kg/m³ ~ 46,7 m³/h
Strömung: 46,7 m³/h / 0,007 m² => 142 m/h => 2,37 m/s
Dynamische Viskosität (150°C, Abgas/Luft): 2,08 * 10^-5 Ns/m²
Reynoldszahl: R = (0,75 kg/m³ * 2,37 m/s * 0,1 m) / (2,08 * 10^-5 Ns/m²) = 8545

Ich hoffe, ich hab nicht irgendwo einen Fehler eingebaut...

Die Reynoldszahl ist zwar recht hoch (hier wird oft 2300 als Übergang genannt, der ist aber fließend und nicht absolut), aber noch weit von einem "Thermomix" entfernt. Daher würde ich davon ausgehen, dass die Kerntemperatur der Strömung auch nach einiger Entfernung und einigen leichten Bögen (<45°) noch relativ nah an der Austrittstemperatur des Kessels ist. Gut abschätzen lässt sich der Einfluss der Kühlung (und der turbulenten Durchmischung) auch daran, wie schnell sich das Abgasrohr im Verhältnis zur angezeigten Temperatur erwärmt. Werde ich vermutlich demnächst mal testen, allerdings sind Gelegenheiten zu dieser Jahreszeit bei mit nicht sehr häufig, da meine Wärmepumpe den gesamten Heiz- und Warmwasserbedarf liefert... Vielleicht provoziere ich es die Tage mal mit dem Kaminkehrer-Modus.

Grüße,
Daniel

PS: Wäre die Stömung hochgradig turbulent, würde die Unterdruckklappe im Abgasrohr meiner Einschätzung nach auch nicht sauber funktionieren...

**Comment by Kenaschon** [comment]
Für die Messung vom Schornsteinfeger hat das Abgasrohr an meiner Heizung auch ein 10mm großes Loch. Das ist ca. 40cm hinter dem "Auslass" vom Kessel. Dort steckt der Sensor einfach drinn. Das Ende des Sensor befindet sich ca. in der Mitte des Abgasrohres. Dort sind es laut Anzeige dann so 165°C.

Ich glaube das passt schon. Ob da jetzt 5 Grad mehr oder weinger sind. Da hau ich mir ein Ei drauf.

**Reply by DonCanDo** [reply]
Moin,

ich wollte mal zum Thema Abgastemperatursensor meinen Senf dazu geben. Weil ich mir unsicher war, ob der *Luft hoch* gemeindeamtliche vorstehende Schornsteinfegermeister oder so ähnlich (manchmal schwierig um Umgang) bei der Feuerstättenschau wegen Eigentümerwechsel Stress macht, hatte ich das Thema mal mit ihm disskutiert und den Sensor erstmal obligatorisch in das existierende Loch vom Heizungsjoe geklebt.

Eigentlich wollte ich das einfach im Blechrohr rein drehen, weil ich dort sowieso schon so ein original ostentativ abgedichtetes Loch habe. Allerdings hatte mir der Schorni dann zum Besten gegeben, dass der Hauptstrom nicht direkt in der Mitte des Kanals sei, sondern etwas weiter außer Mitte und dass der Strom hinter dem Schalldämpfer so dermaßen verwirbelt, dass eine sinnvolle Messung nicht möglich ist. Wo ich ihn dann fragte wo ich den anbringen könne, ohne dass er mir Ärgert macht meinte er schlapp is doch nen Loch da drehst`n da rein (eine Sekunde nachdem er meinen Sensor raus gezupft und seine Sonde dort schon drin hatte.

Damit er mir nicht immer den Sensor raus zieht, werde ich mich mal dran tun und ein neues Loch für ihn auf der zugänglicheren anderen Seite des Rohrs anbringen. Damit das dann auch vernünftig wird, tendiere ich dazu, entweder ein zweites Schornsteinfegerloch auf der anderen Seite zu machen wo man viel besser ran kommt und mir ein Gewinde rein zu schneiden. Oder ich nehme eine Mutter und bringe die da irgendwie an.


Was ich mich aber in dem Zusammenhang noch frage, ist das Thema Abgasüberwachung. Derzeit habe ich nur den Plan, die Abgastemperatur im HA mit zu tracken für eine Szene wenn Brenner an Abgas nicht da dann abschalten oder sowas.

Allerdings hatte ich auch Entitäten rein bekommen über die Abgastemperaturüberwachung der Regelgung. Mit Wert unbekannt. Dort sind 2 von 3 Klemmen auf der Steckerleiste meiner R2107 Reglung einfach gebrückt.

Hat sich damit schon mal jemand beschäftigt, inwiefern man diese aktivieren und nutzen könnte?
Auf dem Bild sind es die Blöcke gelb und blau.

![PXL_20240821_192401031 MP~2 1](https://github.com/user-attachments/assets/1b0adfe0-4113-4288-aa09-4c1f7ad4e5c3)


![PXL_20240924_094433129 MP_1 1](https://github.com/user-attachments/assets/834f7eb4-ddf3-4ba6-8f4b-7808f0d3e35c)

**Reply by the78mole** [reply]
Hallo Don,

tatsächlich weiß ich zur Brücke rein gar nichts. Kann mich auch nicht erinnern, da irgendwas in meinen gesammelten Unterlagen gelesen zu haben. Die Flamme selbst wird ja ohnehin durch den optischen Flammwächter überwacht. Welche Fehlerzustände erwartest Du denn, die eine Abgasüberwachung notwendig machen, die der Flammwächter nicht abfängt? Allenfalls, dass das Abgasrohr vom Kessel abfällt. Eine zu hohe Temperatur? Eher unwahrscheinlich... Die chemische Zusammensetzung kann man ja damit dann auch nicht überwachen... Allenfalls könnte man ein paar wenige Rückschlüsse ziehen, wenn die Düse stark abweichenden Durchfluss hat oder die Zuluft im Volumenstrom stark abweicht.

Grüße,
Daniel

**Comment by FVerhamme** [comment]
Danke für die Infos.

**Comment by DonCanDo** [comment]
Hey Hey,

joah naja ich werde noch mal die Platine umprogrammieren auf Dewennis damit ich die Heizung auch steuern kann. Mit der jetztigen Software kann ich irgendwie nur das Warmwasser steuern und das ESP Update läuft nicht. Ich sehe da noch nicht so richtig durch.

Muss ich mir erstmal alles beibringen.


Von: Daniel Glaser ***@***.***>
Gesendet: Donnerstag, 10. Oktober 2024 14:10
An: dewenni/ESP_Buderus_KM271 ***@***.***>
Cc: Don ***@***.***>; Comment ***@***.***>
Betreff: Re: [dewenni/ESP_Buderus_KM271] Abgastemperatur erfassen? (Discussion #55)


Hallo Don,

tatsächlich weiß ich zur Brücke rein gar nichts. Kann mich auch nicht erinnern, da irgendwas in meinen gesammelten Unterlagen gelesen zu haben. Die Flamme selbst wird ja ohnehin durch den optischen Flammwächter überwacht. Welche Fehlerzustände erwartest Du denn, die eine Abgasüberwachung notwendig machen, die der Flammwächter nicht abfängt? Allenfalls, dass das Abgasrohr vom Kessel abfällt. Eine zu hohe Temperatur? Eher unwahrscheinlich... Die chemische Zusammensetzung kann man ja damit dann auch nicht überwachen... Allenfalls könnte man ein paar wenige Rückschlüsse ziehen, wenn die Düse stark abweichenden Durchfluss hat oder die Zuluft im Volumenstrom stark abweicht.

Grüße,
Daniel

-

**Reply by the78mole** [reply]
Hi Don,

ja, ich muss zugeben, dewennis Firmware finde ich extrem gut. Aber um für ESPhome etwas Support leisten zu können, kann ich schlecht einfach bei mir dewennis FW benutzen...

Grüße,
Daniel

**Comment by DonCanDo** [comment]
Guten Morgen.

Das stimmt letztlich bin ich dir schlichtweg für die Möglichkeit mega dankbar. Ich bin zurzeit ständig fern vom Haus und habe ein viel besseres Gefühl so

Grüße Steven
________________________________
